### PR TITLE
refac(backend-tests): Reduce wait time in login tries to two minues

### PR DESCRIPTION
**Pull request by MaciejTe**
refac(backend-tests): Reduce wait time in login tries to two minutes

Changelog: None
Ticket: None
Signed-off-by: Maciej Tomczuk <maciej.tomczuk@northern.tech>

### CODE REVIEW

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416  --pr useradm-enterprise/517 --fast

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416 --pr useradm-enterprise/517 --fast

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416 --pr useradm-enterprise/517 --fast

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416 --pr useradm-enterprise/517

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416 --pr useradm-enterprise/517

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --pr tenantadm/396 --pr workflows-enterprise/416 --pr useradm-enterprise/517

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --fast

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --fast

File: testutils/common.py [hunk 1]
```diff
@@ -178,15 +178,15 @@ def create_org(
     # Try log in every second for 3 minutes.
     # - There usually is a slight delay (in order of ms) for propagating
     #   the created user to the db.
-    for i in range(3 * 60):
+    for i in range(2 * 60):
```
Comment (lluiscampos) [inline testutils/common.py:181]: Could we use here also a `retrier`?

File: testutils/infra/cli.py [hunk 2]
```diff
@@ -112,10 +112,19 @@ def __init__(self, containers_namespace="backend-tests", container_manager=None)
         BaseCli.__init__(
             self, "mender-tenantadm", containers_namespace, container_manager
         )
+        self.service_name = "mender-tenantadm"
 
-    def create_org(self, name, username, password, plan="os"):
+        open_source = Microservice("/usr/bin/tenantadm", "/etc/tenantadm")
+        enterprise = Microservice(
+            "/usr/bin/tenantadm-enterprise", "/etc/tenantadm-enterprise"
```
Comment (lluiscampos) [inline testutils/infra/cli.py:119]: There is no  `tenantadm-enterprise`.

**Review by lluiscampos — APPROVED** [review]
Thank you.

This PR does a lot more that the commit title says. I am all in favor to do refactoring, cleanups, modernization of code, etc. But I think it should be done in separate, atomic and good described commits.

File: testutils/infra/cli.py (hunk 2 above)
Comment (tranchitella) [inline testutils/infra/cli.py:119]: This is wrong, there are no open-source/enterprise tenantadm variants.

File: backend-tests/tests/test_devauth.py [hunk 3]
```diff
@@ -74,11 +74,11 @@ def user(clean_migrated_mongo):
 
 @pytest.fixture(scope="function")
 def devices(clean_migrated_mongo, user):
-    uc = ApiClient(useradm.URL_MGMT)
+    useradm_mgmt_v1 = ApiClient(useradm.URL_MGMT)
```
Comment (tranchitella) [inline backend-tests/tests/test_devauth.py:77]: I'd split this variable renaming into a dedicated commit.

File: backend-tests/tests/test_tenantadm.py [hunk 4]
```diff
@@ -92,8 +96,8 @@ def test_duplicate_username(self, clean_mongo_tenant_migration):
         )
         self.logger.debug("Tenant id: %s" % tenant_id)
 
-        # Retry login every second for 3 min
-        for i in range(60 * 3):
+        # Retry login every second for 2 min
+        for i in range(60 * 2):
```
Comment (tranchitella) [inline backend-tests/tests/test_tenantadm.py:100]: Sorry for the question: why are we reducing it from 3 mins to 2 mins?
What's the rationale? The commit message doesn't explain it.

File: backend-tests/tests/test_useradm_enterprise.py [hunk 5]
```diff
@@ -122,41 +119,66 @@ def _make_2fa_settings(self, user_statuses):
...
+    @pytest.fixture(scope="function")
+    def setup_user_no_2fa(self, tenants_users) -> User:
+        user_no_2fa = tenants_users[0].users[1]
 
+        r = self._login(user_no_2fa)
+        assert r.status_code == 200
+        user_no_2fa.token = r.text
+        yield user_no_2fa
+
+    def test_2fa_cannot_be_enabled_disabled_before_email_verification(
```
Comment (tranchitella) [inline backend-tests/tests/test_useradm_enterprise.py:140]: How is this related to the reduced wait time in the login?
In other words, what are you fixing here?

File: testutils/infra/cli.py (hunk 2 above)
Comment (MaciejTe) [reply testutils/infra/cli.py]: Oops! Fixed.

File: backend-tests/tests/test_devauth.py (hunk 3 above)
Comment (MaciejTe) [reply backend-tests/tests/test_devauth.py:77]: Done!

File: backend-tests/tests/test_tenantadm.py (hunk 4 above)
Comment (MaciejTe) [reply backend-tests/tests/test_tenantadm.py:100]: Saving user to the database does not take 3 minutes for sure, I'm trying to save some time in frequently running tests.

File: backend-tests/tests/test_useradm_enterprise.py (hunk 5 above)
Comment (MaciejTe) [reply backend-tests/tests/test_useradm_enterprise.py:140]: I created separate commit for that.

File: testutils/common.py (hunk 1 above)
Comment (MaciejTe) [reply testutils/common.py]: Done!

**Comment by MaciejTe** [comment]
@mender-test-bot start pipeline --fast

**Review by tranchitella — APPROVED** [review]
Commit 7bb0b60adac49f9d12400bfe0dd131d503977bd4 has some typos, both in the title and in the description.
